lib/pipe/html.js

```diff
@@ -209,42 +221,76 @@ class HtmlPipe {
     );
 
-    handlebars.registerHelper('json', tests => {
-      // TODO: please refactor the code below, add meaningful variable names, separate into functions
-      function replaceScriptTagsInArray(array) {
-        return array.map(obj => {
-          const keysToCheck = ['steps', 'stack', 'title', 'suite_title', 'message', 'code'];
-          const newObj = {};
-
-          for (const key in obj) {
-            if (Object.prototype.hasOwnProperty.call(obj, key)) {
-              if (key === 'example') {
-                newObj[key] = {};
-                for (const subKey in obj[key]) {
-                  if (Object.prototype.hasOwnProperty.call(obj[key], subKey)) {
-                    newObj[key][subKey] =
-                      typeof obj[key][subKey] === 'string'
-                        ? obj[key][subKey].replace(/<script>/g, '<$cript>').replace(/<\/script>/g, '</$cript>')
-                        : obj[key][subKey];
-                  }
-                }
-              } else if (keysToCheck.includes(key)) {
-                newObj[key] =
-                  typeof obj[key] === 'string'
-                    ? obj[key].replace(/<script>/g, '<$cript>').replace(/<\/script>/g, '</$cript>')
-                    : obj[key];
-              } else {
-                newObj[key] = obj[key];
-              }
+    handlebars.registerHelper(
+      'selectComponent',
+      () =>
+        new handlebars.SafeString(
+          `<select style="width: 70px;height: 38px;" class="form-select" aria-label="Tests counter on page">
+          <option value="0">10</option>
+          <option value="1">25</option>
+          <option value="2">50</option>
+        </select>`,
+        ),
+    );
+
+    handlebars.registerHelper(
+      'emptyDataComponent',
+      () =>
+        new handlebars.SafeString(
+          `<div class="noData m-0">
+          NO MATCHING TESTS
+        </div>`,
+        ),
+    );
+
+    handlebars.registerHelper('pageDispleyElements', tests => {
+      // We wrapp the lines to the HTML format we need
+      const totalTests = JSON.parse(
+        JSON.stringify(tests)
+          .replace(/<script>/g, '&lt;script&gt;')
+          .replace(/<\/script>/g, '&lt;/script&gt;'), // eslint-disable-line
+      );
+
+      const paginationOptions = {
+        0: 10,
+        1: 25,
+        2: 50,
+      };
+      const statuses = ['all', 'passed', 'failed', 'skipped'];
+      const pageItemGroups = {
+        all: {},
+        passed: {},
+        failed: {},
+        skipped: {},
+        totalTests,
+      };
+
+      function paginateItems(items, pageSize) {
+        const paginatedItems = [];
+        for (let i = 0; i < items.length; i += pageSize) {
+          paginatedItems.push(items.slice(i, i + pageSize));
+        }
+        return paginatedItems;
+      }
+
+      statuses.forEach(status => {
+        for (const option in paginationOptions) {
+          if (Object.hasOwnProperty.call(paginationOptions, option)) {
```

Comment:
`Object.hasOwnProperty.call` ??
why not `paginationOptions.hasOwnProperty(option)` ?

Reply:
because of linter, but I fixed this line to